torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -74,6 +74,15 @@ def embedding_dense_backward_rules(op_schema: OpSchema) -> OutputSharding:
         return OutputSharding(
             output_spec=DTensorSpec(mesh=indices.mesh, placements=[_Partial()])
         )
+    elif (
+        grad_output.placements[0].is_partial() and indices.placements[0].is_replicate()
+    ):
+        # The embedding table is replicated and the indices is also replicated
```

Comment:
hmm, why indices are also replicated?

And this assumes the placements list only contains one element?

Reply:
The code is like `pos = pos_emb(inp.size()[1])` where `pos_emb` is a nn.Embedding. It is not related to batch dimension so it is replicated (local may be more precise).